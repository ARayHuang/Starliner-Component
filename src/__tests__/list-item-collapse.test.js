import React from 'react';
import { shallow, mount } from 'enzyme';
import ListItemCollapse from '../components/list/list-item-collapse';

describe('ListItemCollapse', () => {
	it('should handle default props', () => {
		const {
			isLink,
			classes,
			onClick,
		} = ListItemCollapse.defaultProps;

		const expectClasses = {
			collapseTitle: '',
			listItem: '',
			selected: ''
		};

		expect(isLink).toBe(false);
		expect(classes).toEqual(expectClasses);
		expect(onClick).toBeDefined();
		expect(onClick).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			classes: {
				collapseTitle: '',
				listItem: '',
				selected: ''
			},
			text: 'collapse-text',
			selectedKey: 'mock-selected-key',
			subItems: [
				{
					text: 'item-1',
					key: 'item-2',
					to: 'goto-item-1',
					onClick: () => {},
					isSelected: true
				},
				{
					text: 'item-2',
					key: 'item-2',
					to: 'goto-item-2',
					onClick: () => {},
					isSelected: false
				}
			]
		};

		const wrapper = shallow(
			<ListItemCollapse {...props} />
		)

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const props = {
			classes: {
				collapseTitle: '',
				listItem: '',
				selected: ''
			},
			text: 'collapse-text',
			selectedKey: 'mock-selected-key',
			subItems: [
				{
					key: 'item-1',
					text: 'item-1',
					to: 'goto-item-1',
					onClick: () => {},
					isSelected: true
				},
				{
					key: 'item-2',
					text: 'item-2',
					to: 'goto-item-2',
					onClick: () => {},
					isSelected: false
				}
			]
		};
		const wrapper = mount(
			<ListItemCollapse {...props} />
		);

		expect(wrapper.props().classes).toEqual(props.classes);
		expect(wrapper.props().text).toEqual(props.text);
		expect(wrapper.props().selectedKey).toEqual(props.selectedKey);
		expect(wrapper.props().subItems).toEqual(props.subItems);
	});

	// it('should handle onClick event', () => {
	// 	const onClickMock = jest.fn();

	// 	const wrapper = mount(
	// 		<ListItem onClick={onClickMock}>
	// 			<div>mock-children</div>
	// 		</ListItem>
	// 	);

	// 	wrapper.simulate('click');

	// 	expect(onClickMock).toHaveBeenCalled();
	// 	expect(onClickMock).toHaveBeenCalledTimes(1);
	// });
})
