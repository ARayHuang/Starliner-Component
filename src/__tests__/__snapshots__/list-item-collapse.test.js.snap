// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemCollapse should render correctly 1`] = `
<Fragment>
  <ListItem
    alignItems="center"
    button={true}
    className=""
    classes={
      Object {
        "selected": "",
      }
    }
    hasBottomDivider={false}
    hasItemPadding={true}
    isButton={false}
    isDense={true}
    isDisabled={false}
    isSelected={false}
    onClick={[Function]}
  >
    <div
      className=""
    >
      collapse-text
    </div>
    <Memo(ForwardRef(ExpandMoreIcon)) />
  </ListItem>
  <WithStyles(ForwardRef(Collapse))
    in={false}
    timeout="auto"
    unmountOnExit={true}
  >
    <List
      className=""
      component="div"
      disablePadding={true}
    >
      <ListItem
        alignItems="center"
        className="makeStyles-defaultSubItem-1 "
        classes={
          Object {
            "selected": "",
          }
        }
        hasBottomDivider={false}
        hasItemPadding={true}
        isButton={true}
        isDense={true}
        isDisabled={false}
        isSelected={false}
        key="item-2"
        onClick={[Function]}
      >
        item-1
      </ListItem>
      <ListItem
        alignItems="center"
        className="makeStyles-defaultSubItem-1 "
        classes={
          Object {
            "selected": "",
          }
        }
        hasBottomDivider={false}
        hasItemPadding={true}
        isButton={true}
        isDense={true}
        isDisabled={false}
        isSelected={false}
        key="item-2"
        onClick={[Function]}
      >
        item-2
      </ListItem>
    </List>
  </WithStyles(ForwardRef(Collapse))>
</Fragment>
`;
